himbaechel/uarch/gowin/globals.cc

```diff
@@ -795,23 +795,44 @@ struct GowinGlobalRouter
         if (ctx->debug) {
             log_info("    step 3: %s -> \n", ctx->nameOfWire(src_wire));
         }
+        // Create a temporary small network where segment gates will be the sinks
+        IdString gate_net_name = ctx->id(ni->name.str(ctx) + std::string("$gate_net$"));
```

Comment:
this would be neater using `ctx->idf("%s$gate_net$", ni->name.c_str(ctx))`

Reply:
Done.